include/openenclave/attestation/attester.h

```diff
@@ -0,0 +1,154 @@
+
+#error "The header attester.h is only available for the enclave."
+
+#else // OE_BUILD_ENCLAVE
+
+#include <openenclave/bits/report.h>
+#include <openenclave/bits/result.h>
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * oe_attester_initialize
+ *
+ * Initializes the attester environment configured for the platform and
+ * the calling application.
+ * This function is only available in the enclave.
+ *
+ * This function is idempotent and can be called multiple times without
+ * adverse effect.
+ *
+ * @experimental
+ *
+ * @retval OE_OK on success.
+ * @retval other appropriate error code.
+ */
+oe_result_t oe_attester_initialize(void);
+
+/**
+ * oe_attester_select_format
+ *
+ * Selects the left-most evidence format from the input list that is supported.
+ * This function is only available in the enclave.
+ *
+ * @experimental
+ *
+ * @param[in] formats Ordered list of evidence formats from which to
+ * select, with descending priority order from left to right.
+ * @param[in] formats_length The number of entries in the input evidence
+ * format list.
+ * @param[out] selected_format Pointer to a caller-supplied buffer to
+ * hold the selected evidence format.
+ * @retval OE_OK on success.
+ * @retval OE_INVALID_PARAMTER At least one of the parameters is invalid.
+ * @retval OE_NOT_FOUND if no format in the input list is supported.
+ * @retval other appropriate error code.
+ */
+oe_result_t oe_attester_select_format(
+    const oe_uuid_t* formats,
+    size_t formats_length,
+    oe_uuid_t* selected_format);
+
+/**
+ * oe_get_evidence
+ *
+ * Generates the evidence for the given format UUID.
+ * This function is only available in the enclave.
+ *
+ * @experimental
+ *
+ * @param[in] evidence_format The format UUID of the evidence to be generated.
+ * @param[in] custom_claims The optional custom claims list.
+ * @param[in] custom_claims_length The number of custom claims.
+ * @param[in] opt_params The optional format-specific input parameters.
+ * @param[in] opt_params_size The size of opt_params in bytes.
+ * @param[out] evidence_buffer An output pointer that will be assigned the
+ * address of the dynamically allocated evidence buffer.
+ * @param[out] evidence_buffer_size A pointer that points to the size of the
+ * evidence buffer in bytes.
+ * @param[out] endorsements_buffer An output pointer that will be assigned the
+ * address of the  dynamically allocated endorsements buffer.
+ * @param[out] endorsements_buffer_size A pointer that points to the size of the
+ * endorsements buffer in bytes.
+ * @retval OE_OK The function succeeded.
+ * @retval OE_INVALID_PARAMTER At least one of the parameters is invalid.
+ * @retval OE_NOT_FOUND The input evidence format is not supported.
+ * @retval other appropriate error code.
+ */
+oe_result_t oe_get_evidence(
+    const oe_uuid_t* evidence_format,
+    const oe_claim_t* custom_claims,
+    size_t custom_claims_length,
+    const void* opt_params,
+    size_t opt_params_size,
```

Comment:
Guidelines for public APIs say to not use abbreviations.
```suggestion
    const void* optional_parameters,
    size_t optional_parameters_size,
```

Reply:
Thanks!